From 27b39257ba2f86e566e935d0b71e6097162c6172 Mon Sep 17 00:00:00 2001
From: Changcheng Liu <changcheng.liu@intel.com>
Date: Wed, 16 Jan 2019 19:54:17 +0800
Subject: [PATCH 02/38] give service endpoint to access to RESTful plugin

endpoint is the base url address to acccess to
RESTful plugin servie. It'll be given by caller or
use the default address.

RESTful plugin service address may not be localhost

Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>
---
 cephclient/client.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/cephclient/client.py b/cephclient/client.py
index 4de0337..dcb9e7a 100644
--- a/cephclient/client.py
+++ b/cephclient/client.py
@@ -53,7 +53,12 @@ class CephClient(object):
 
         self.log.debug("Params: {0}".format(str(self.params)))
 
-        self.endpoint = self.params['endpoint']
+        if 'endpoint' in self.params:
+            self.endpoint = self.params['endpoint']
+        else:
+            # default endpoint
+            self.endpoint = 'https://localhost:5001/'
+
         if 'timeout' not in self.params:
             self.timeout = None
 
-- 
2.17.1

